coalib/results/TextPosition.py

```diff
@@ -7,20 +7,35 @@
 class TextPosition:
 
     @enforce_signature
-    def __init__(self, line: (int, None)=None, column: (int, None)=None):
+    def __init__(self,
+                 line: (int, None)=None,
+                 column: (int, None)=None,
+                 text: (tuple, list, None)=None,
```

Comment:
An object is also used for structured data. A text position is completely defined by just a position. Now a function can be made later which takes a TextPosition and only needs position (it doesn't really need line or column or text).

Hence its a valid use case which doesn't use all helper functions in the class - but it defines a TextPosition none the less.

-----Original Message-----
From: "Lasse Schuirmann" <notifications@github.com>
Sent: ‎27-‎03-‎2016 07:12
To: "coala-analyzer/coala" <coala@noreply.github.com>
Cc: "AbdealiJK" <abdealikothari@gmail.com>
Subject: Re: [coala] SourceRange add position functionalities  (#1947)

In coalib/results/TextPosition.py:
> @@ -7,20 +7,35 @@
>  class TextPosition:
>  
>      @enforce_signature
> -    def __init__(self, line: (int, None)=None, column: (int, None)=None):
> +    def __init__(self,
> +                 line: (int, None)=None,
> +                 column: (int, None)=None,
> +                 text: (tuple, list, None)=None,
if the user gives position only, there's no way the other information can be derived. The whole point of having an object is so we can rely on it doing certain things for us, you remove that value and the object may be of no use for us and we'd need to do checks for every information which sucks. We have an object which is unusable in some cases.
—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub